pkg/clients/oss/oss.go

```diff
@@ -0,0 +1,204 @@
+type ClientInterface interface {
+	Describe(name string) (*sdk.GetBucketInfoResult, error)
+	Create(bucket v1alpha1.BucketParameter) error
+	Update(name string, aclStr string) error
+	Delete(name string) error
+}
+
+// SDKClient is the SDK client for Bucket
+type SDKClient struct {
+	Client *sdk.Client
+}
+
+// NewClient will create Bucket client
+func NewClient(ctx context.Context, endpoint string, accessKeyID string, accessKeySecret string, stsToken string) (*SDKClient, error) {
+	var (
+		client *sdk.Client
+		err    error
+	)
+	if stsToken == "" {
+		client, err = sdk.New(endpoint, accessKeyID, accessKeySecret)
+	} else {
+		client, err = sdk.New(endpoint, accessKeyID, accessKeySecret, sdk.SecurityToken(stsToken))
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to crate Bucket client: %v", err)
+	}
+	klog.Info("successfully created Bucket client.")
+	return &SDKClient{Client: client}, nil
+}
+
+// Describe describes Bucket bucket
+func (c *SDKClient) Describe(name string) (*sdk.GetBucketInfoResult, error) {
+	bucketInfoResult, err := c.Client.GetBucketInfo(name)
+	if err != nil {
+		klog.ErrorS(err, "failed to get bucket information", "Bucket", name)
+		return nil, err
+	}
+	return &bucketInfoResult, nil
+}
+
+// Create creates Bucket bucket
+func (c *SDKClient) Create(bucket v1alpha1.BucketParameter) error {
+	var options []sdk.Option
+	var (
+		acl                sdk.ACLType
+		storageClass       sdk.StorageClassType
+		dataRedundancyType sdk.DataRedundancyType
+		err                error
+	)
+
+	// validate ACL
+	acl, err = ValidateOSSAcl(bucket.ACL)
+	if err != nil {
+		return err
+	}
+	options = append(options, sdk.ACL(acl))
+
+	// validate StorageClass
+	storageClass, err = validateOSSStorageClass(bucket.StorageClass)
+	if err != nil {
+		klog.Error(err)
+		return err
+	}
+	options = append(options, sdk.StorageClass(storageClass))
+
+	// validate DataRedundancyType
+	dataRedundancyType, err = validateOSSDataRedundancyType(bucket.DataRedundancyType)
+	if err != nil {
+		klog.Error(err)
+		return err
+	}
+	options = append(options, sdk.RedundancyType(dataRedundancyType))
+
+	if err := c.Client.CreateBucket(bucket.Name, options...); err != nil {
+		klog.ErrorS(err, "failed to create Bucket bucket", "Bucket", bucket.Name)
+		return err
+	}
+	return nil
+}
+
+// Update sets bucket acl
+func (c *SDKClient) Update(name string, aclStr string) error {
+	acl, err := ValidateOSSAcl(aclStr)
+	if err != nil {
+		klog.ErrorS(err, "Name", name, "ACL", aclStr)
+		return err
+	}
+	return c.Client.SetBucketACL(name, acl)
+}
+
+// Delete deletes SLS project
+func (c *SDKClient) Delete(name string) error {
+	klog.InfoS("deleting Bucket bucket", "Name", name)
+	return c.Client.DeleteBucket(name)
+}
+
+// IsNotFoundError checks whether the error is an NotFound error
+func IsNotFoundError(err error) bool {
+	if err == nil {
+		return false
+	}
+	e, ok := err.(sdk.ServiceError)
+	return ok && e.Code == "NoSuchBucket"
```

Comment:
Consider using const or SDK var for `NoSuchBucket`.

Reply:
Sure. Done.

Though it's used in one place, we should use a const for it? Thanks.